pkg/apis/apps/v1beta1/types.go

```diff
@@ -330,6 +330,12 @@ type DeploymentStatus struct {
 	// +patchMergeKey=type
 	// +patchStrategy=merge
 	Conditions []DeploymentCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,6,rep,name=conditions"`
+
+	// Monotonically increasing counter that tracks hash collisions for
+	// the Deployment. Used as a collision avoidance mechanism by the
+	// Deployment controller.
+	// +optional
+	Uniquifier *int64 `json:"uniquifier,omitempty" protobuf:"varint,8,opt,name=uniquifier"`
```

Comment:
Putting it in status was discussed in the proposal for hash collision. We don't want users to set it or think about it. 

API server can update status as well (if we implement it).